src/hooks/useApproveCallback.ts

```diff
@@ -100,22 +101,21 @@ export function useApproveCallback(
 }
 
 // wraps useApproveCallback in the context of a swap
-export function useApproveCallbackFromTrade(trade?: Trade, allowedSlippage = 0) {
-  const { chainId } = useActiveWeb3React()
+export function useApproveCallbackFromTrade(chainId: ChainId, trade?: Trade, allowedSlippage = 0) {
   const amountToApprove = useMemo(
-    () => (trade ? computeSlippageAdjustedAmounts(trade, allowedSlippage)[Field.INPUT] : undefined),
-    [trade, allowedSlippage]
+    () => (trade ? computeSlippageAdjustedAmounts(trade, allowedSlippage, chainId? chainId : ChainId.AVALANCHE)[Field.INPUT] : undefined),
+    [trade, allowedSlippage, chainId]
   )
-  return useApproveCallback(amountToApprove, chainId ? ROUTER_ADDRESS[chainId] : ROUTER_ADDRESS[ChainId.AVALANCHE])
+  return useApproveCallback(chainId ? chainId : ChainId.AVALANCHE, amountToApprove, chainId ? ROUTER_ADDRESS[chainId] : ROUTER_ADDRESS[ChainId.AVALANCHE])
 }
 
 // wraps useApproveCallback in the context of a swap
-export function useApproveCallbackFromInputCurrencyAmount(currencyAmountIn: any | undefined) {
+export function useApproveCallbackFromInputCurrencyAmount(chainId: ChainId, currencyAmountIn: any | undefined) {
```

Comment:
can we use directly `const { chainId } = useActiveWeb3React()` instead of passing chainId

Reply:
Resolved,
you are right, way better this way.